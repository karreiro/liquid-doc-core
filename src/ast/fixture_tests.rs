// In your test file
include!(concat!(env!("OUT_DIR"), "/fixtures_macro.rs"));

macro_rules! generate_fixture_test {
    ($fixture_name:expr) => {
        paste::paste! {
            #[test]
            fn [<test_fixture_ $fixture_name>]() {
                let content = include_str!(concat!("../../web/fixtures/", $fixture_name, ".liquid"));
                // Your test logic here
                crate::assert_json_output!(content);
            }
        }
    };
}

// This will generate all the tests
for_each_fixture_file!(generate_fixture_test);
