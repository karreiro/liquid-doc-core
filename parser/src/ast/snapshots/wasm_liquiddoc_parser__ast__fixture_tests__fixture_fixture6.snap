---
source: src/ast/fixture_tests.rs
expression: "@description This is a description\n@example This is an example\n@param {String} paramWithDescription - param with description"
---
[
  {
    "type": "LiquidDocDescriptionNode",
    "name": "description",
    "position": {
      "start": 10,
      "end": 45
    },
    "source": "@description This is a description\n",
    "content": {
      "type": "TextNode",
      "value": "This is a description\n",
      "position": {
        "start": 23,
        "end": 45
      },
      "source": "@description This is a description\n"
    },
    "isImplicit": false,
    "isInline": true
  },
  {
    "type": "LiquidDocExampleNode",
    "name": "example",
    "position": {
      "start": 45,
      "end": 73
    },
    "source": "@example This is an example\n",
    "content": {
      "type": "TextNode",
      "value": "This is an example\n",
      "position": {
        "start": 54,
        "end": 73
      },
      "source": "@example This is an example\n"
    },
    "isInline": true
  },
  {
    "type": "LiquidDocParamNode",
    "name": "param",
    "position": {
      "start": 73,
      "end": 134
    },
    "source": "@param {String} paramWithDescription - param with description",
    "paramName": {
      "type": "TextNode",
      "value": "paramWithDescription",
      "position": {
        "start": 89,
        "end": 109
      },
      "source": "paramWithDescription"
    },
    "paramDescription": {
      "type": "TextNode",
      "value": "param with description",
      "position": {
        "start": 112,
        "end": 134
      },
      "source": "param with description"
    },
    "paramType": {
      "type": "TextNode",
      "value": "String",
      "position": {
        "start": 81,
        "end": 87
      },
      "source": "{String}"
    },
    "required": true
  }
]
